src/libmongoc/tests/test-mongoc-client.c

```diff
@@ -3341,7 +3370,7 @@ _test_client_sends_handshake (bool pooled)
    request_destroy (request);
 
    if (!pooled) {
-      _respond_to_ping (future, server);
+      _respond_to_ping (future, server, false);
```

Comment:
Running the test locally suggests the 3rd argument should be `true`. `mongoc_client_uses_server_api(client)` returns `false`.